certbot/hooks.py

```diff
@@ -151,3 +247,15 @@ def execute(shell_cmd):
         logger.error('Error output from %s:\n%s', base_cmd, err)
     return (err, out)
 
+
+def list_hooks(dir_path):
+    """List paths to all hooks found in dir_path in sorted order.
+
+    :param str dir_path: directory to search
+
+    :returns: `list` of `str`
+    :rtype: sorted list of absolute paths to executables in dir_path
+
+    """
+    paths = (os.path.join(dir_path, f) for f in os.listdir(dir_path))
+    return sorted(path for path in paths if util.is_exe(path))
```

Comment:
Interesting. Googling around, I found [this](https://www.gnu.org/software/coreutils/faq/coreutils-faq.html#The-ls-command-is-not-listing-files-in-a-normal-order_0021) which explains that sorting order is based on your locale. While I think sorting here is only a nicety and not essential (if order really matters, you can wrap the commands in a script in the order you want them run), it'd be kind of nice if we had this behavior.

In seeing how we can do this in Python, I saw the first bullet point [here](https://docs.python.org/3/howto/sorting.html#odd-and-ends). Unfortunately, at least testing this on a couple Ubuntu machines, neither approach described there works for me when using the locale as configured by Ubuntu/Debian tooling on a couple different machines. (The `locale` shell command works, but the methods in the locale module do not. `locale.getlocale()` returns `(None, None)`.)

If you think this a peculiarity with my systems and this would work in almost all other places (or you know of another approach), I'd be happy to make this change. Barring that though, I think we should just stick with the current sorting order and I'll update the documentation to clarify this minor point.

Reply:
Ah! I wasn't aware that it's because of the locales. TIL.

I played around with it a bit, and we would be able to something like:
```
import locale
if not locale.getlocale(locale.LC_COLLATE)[0]:
    c_loc = ".".join(locale.getdefaultlocale())
    locale.setlocale(locale.LC_COLLATE, c_loc)

sorted(our_list, cmp=locale.strcoll)
```
That would check if there's setting for LC_COLLATE, and if not, it would be dug from default locale.

Reply:
After reading [this section of the locale documentation](https://docs.python.org/2/library/locale.html#background-details-hints-tips-and-caveats), I wonder if we shouldn't just call `locale.setlocale(locale.LC_ALL, '')` early in Certbot's execution.

@erikrose, do you have experience with this stuff?

Reply:
I think the first paragraph in linked document is enough to skip doing this and go with what default `sorted()` gives us;
>The C standard defines the locale as a program-wide property that may be relatively expensive to change. On top of that, some implementation are broken in such a way that frequent locale changes may cause core dumps. This makes the locale somewhat painful to use correctly.

Reply:
@bmw I haven't mucked with it in my memory. It sounds like a world of hurt, though, and I'd be keen to avoid it! :-)

Reply:
OK. While I think it'd be nice to run these in sorted order based on the user's locale (and don't feel good about suddenly changing this behavior later), I agree the risks probably outweigh the rewards. Let's skip doing this and I'll make a minor note that the sorted order is not locale aware and is based on the numerical value of the characters in the file names.